packages/client/hmi-client/src/views/DocumentView.vue

```diff
@@ -52,6 +61,28 @@ function setActiveTab(index: number) {
 	tabStore.setActiveTabIndex(tabContext, index);
 }
 
+const fetchAnnotations = async () => {
+	const response = await API.get('/annotations', {
+		params: {
+			artifact_type: ResourceType.XDD,
+			artifact_id: newDocumentId.value
+		}
+	});
+	annotations.value = response.data;
+};
+
+const addAnnotation = async () => {
+	const content = annotationContent.value;
+	await API.post('/annotations', {
+		content,
+		artifact_id: newDocumentId.value,
+		artifact_type: ResourceType.XDD // FIXME: need to establish artifact types
+	});
+
+	// Refresh
+	await fetchAnnotations();
+};
```

Comment:
Those should be in a service.

Reply:
Given how messy the service architecture is I will just add a FIXME for now.